🐛 Hide placeholder when tweet loads successfully

kvchari opened the pull request:
fixes #34616 

<!--
# Instructions:

1. Pick a meaningful title for your pull request.
  a. Prefix the title with an emoji. (Copy-paste from the list below.)
  b. If helpful, use a short prefix (e.g. `[Project XX] Implement feature YY`).
2. Enter a description that explains why the PR is necessary, and what it does.
  a. Mention the GitHub issue being addressed by this pull request.
  b. Use keywords to auto-close linked issues during merge. (e.g. `Fixes #11111`, or `Closes #22222`)
3. For substantial changes, first file an Intent-to-Implement (I2I) issue at go.amp.dev/i2i.

# References:

- AMP code contribution docs: go.amp.dev/contribute/code
- First time setup (required for CI checks): go.amp.dev/getting-started

# Emojis for categorizing pull requests (copy-paste emoji into description):

✨ New feature
🐛 Bug fix
🔥 P0 fix
✅ Tests
❄️ Flaky tests
🚀 Performance improvements
🖍 CSS / Styling
♿ Accessibility
🌐 Internationalization
📖 Documentation
🏗 Infrastructure / Tooling / Builds / CI
⏪ Revert
♻️ Refactor
🚮 Deletion
🧪 Experimental code
-->

rcebulko commented on extensions/amp-twitter/1.0/component.js:
```diff
@@ -45,9 +46,13 @@ function TwitterWithRef(
           setHeight(height);
         }
+
+        if (togglePlaceholder) {
+          togglePlaceholder(false);
+        }
```
```suggestion
        togglePlaceholder?.(false);
```

kvchari replied:
updated

rcebulko reviewed (commented):
Should this PR contain the `tsconfig.json` changes?

kvchari commented:
@rcebulko rebased to drop the tsconfig commits

kvchari commented:
removed the @types/enzyme dependency i added because multiple typings for react end up getting installed causing type errors (https://app.circleci.com/pipelines/github/ampproject/amphtml/15289/workflows/bf7a8dd0-de3b-49bd-a077-05f45325ed02/jobs/264732). resolving to 1 version would solve this, but is out of scope for this

caroqliu commented on extensions/amp-twitter/1.0/component.js:
```diff
@@ -27,6 +27,7 @@ function TwitterWithRef(
     timelineSourceType,
     timelineUserId,
     title,
+    togglePlaceholder,
```
While this approach works, the concept of a placeholder is somewhat specific to AMP, and we don't need to be as prescriptive as to assume that an author wants to implement the component this way.

I'd recommend that the Preact layer have a more generic prop, such as `onLoad` and `onError`, which are called at appropriate lifecycle states -- perhaps `onLoad` once the `height` is received and `onError` when receiving a [`NO_CONTENT` message like in `0.1`](https://github.com/ampproject/amphtml/blob/24174aa4c722d020f276da6e0bf5f64f0e788b73/extensions/amp-twitter/0.1/amp-twitter.js#L100). And then the AMP layer would pass the placeholder accordingly:
```js
// amp-twitter.js
onError: () => this.togglePlaceholder(true),
onLoad: () => this.togglePlaceholder(false)
```

kvchari replied:
i like this approach better - updated to use do that
